models/mongo.models.go

```diff
@@ -0,0 +1,119 @@
+		Password string `bson:"password" json:"password" binding:"required"`
+	}
+
+	// RegisterForm struct a form for register a Tyr User.
+	RegisterForm struct {
+		Email                string `bson:"email" json:"email" binding:"required"`
+		Password             string `bson:"password" json:"password" binding:"required"`
+		PasswordConfirmation string `bson:"passwordConfirmation" json:"passwordConfirmation" binding:"required"`
+		First                string `bson:"firstName" json:"firstName" binding:"required"`
+		Last                 string `bson:"lastName" json:"lastName" binding:"required"`
+	}
+
+	// FacingTests struct to store the number of tests for private and public facing tests.
+	FacingTests struct {
+		Pass int `bson:"pass" json:"pass" binding:"required"`
+		Fail int `bson:"fail" json:"fail" binding:"required"`
+	}
+
+	// Cases struct to store tests/failed passed for admin/student
+	Cases struct {
+		StudentFacing FacingTests `bson:"studentFacing" json:"studentFacing" binding:"required"`
+		AdminFacing   FacingTests `bson:"adminFacing" json:"adminFacing" binding:"required"`
+	}
+
+	// Submission struct the struct to represent a submission to an page.
+	Submission struct {
+		ID             bson.ObjectId       `bson:"_id" json:"id" binding:"required"`
+		UserID         bson.ObjectId       `bson:"userID" json:"userID" binding:"required"`
+		AttemptNumber  int                 `bson:"attemptNumber" json:"attemptNumber" binding:"required"`
+		SubmissionDate bson.MongoTimestamp `bson:"submissionDate" json:"submissionDate" binding:"required"`
+		File           string              `bson:"file" json:"file" binding:"required"`
+		ErrorTesting   bool                `bson:"errorTesting" json:"errorTesting" binding:"required"`
+		Cases          Cases               `bson:"cases" json:"cases" binding:"required"`
+	}
+
+	// TestScripts struct to represent filenames on gcp storage of scripts.
+	TestScripts struct {
+		StudentFacing string `bson:"studentFacing" json:"studentFacing" binding:"required"`
+		AdminFacing   string `bson:"adminFacing" json:"adminFacing" binding:"required"`
+	}
+
+	AssignmentSubmission struct {
+		UserID        bson.ObjectId `bson:"userID" json:"userID" binding:"required"`
+		SubmissionID  bson.ObjectId `bson:"submissionID" json:"submissionID" binding:"required"`
+		AttemptNumber int           `bson:"attemptNumber" json:"attemptNumber" binding:"required"`
+	}
+
+	// Assignment struct to store information about an assignment.
+	Assignment struct {
+		ID              bson.ObjectId          `bson:"_id" json:"id" binding:"required"`
+		Language        string                 `bson:"language" json:"lanaguage" binding:"required"`
+		Version         string                 `bson:"version" json:"version" binding:"required"`
+		Name            string                 `bson:"name" json:"name" binding:"required"`
+		NumAttempts     int                    `bson:"numAttempts" json:"numAttempts" binding:"required"`
```

Comment:
We should have 2 "NumAttempts", one for the max and one for the current number of attempts, probably named "TimesAttempted" and "MaxAttempts" or something like that

Reply:
The Assignment != a Submission. An Assignment is for everyone so if we increment the number of times attempted per student this would cause problems.

Reply:
Assignment also has  a reference to submissions tables and will know this information.

Reply:
Ok my b